app/controllers/api/proxy_rules_controller.rb

```diff
@@ -0,0 +1,67 @@
+class Api::ProxyRulesController < Api::BaseController
+  before_action :authorize_rolling_update, :find_service, :find_proxy
+
+  activate_menu :serviceadmin, :integration, :mapping_rules
+
+  sublayout 'api/service'
+
+  def index
+    @proxy_rules = @proxy.proxy_rules.includes(:metric).ordered.paginate(page: params[:page])
+  end
+
+  def new
+    last_position = @proxy.proxy_rules.maximum(:position) || 0
+    next_position = last_position + 1
+    @proxy_rule = @proxy.proxy_rules.build(position: next_position)
+  end
+
+  def create
+    @proxy_rule = @proxy.proxy_rules.build(proxy_rule_params)
+    if @proxy_rule.save
+      redirect_to admin_service_proxy_rules_path(@service), notice: 'Mapping rule was created.'
+    else
+      render :new
+    end
+  end
+
+  def edit
+    @proxy_rule = @proxy.proxy_rules.find(params[:id])
+  end
+
+  def update
+    @proxy_rule = @proxy.proxy_rules.find(params[:id])
+    if @proxy_rule.update_attributes(proxy_rule_params)
+      redirect_to admin_service_proxy_rules_path(@service), notice: 'Mapping rule was updated.'
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    if @proxy.proxy_rules.find(params[:id]).destroy
+      flash[:notice] = 'The mapping rule was deleted'
+    else
+      flash[:error] = 'The mapping rule cannot be deleted'
+    end
+
+    redirect_to admin_service_proxy_rules_path(@service)
+  end
+
+  private
+
+  def authorize_rolling_update
+    provider_can_use!(:independent_mapping_rules)
+  end
+
+  def find_proxy
+    @proxy = @service.proxy
```

Comment:
`@service` can be nil and then this would raise an error for sure

Reply:
@Martouta I think we should always rely on service.. wdyt on moving the `find_by` from find_service to `find` ?

with this will throw the exception if service does not exists

Reply:
yes, i think it should be `find` and we rely on its existence 😄 

I also think that you should add an integration test for this controller 👍